internal/edge/key.go

```diff
@@ -20,15 +20,26 @@ type edgeKey struct {
 
 // SetKey parses and associates an Edge key to the agent.
 // If the agent is running inside a Swarm cluster, it will also set the "set" flag to specify that a key is set on this agent in the cluster.
+// Don't overwrite the file if it exists - this only appears to work, because our container runs as root
```

Comment:
I think the changes to this file can be saved for the edge refactor PR, no?

and for that, IMO just rewrite the file (and save it as rewritable)